codegen/templates/service_mock.tmpl

```diff
@@ -0,0 +1,160 @@
+{{- $instance := . -}}
+{{- $leafClass := index .DependencyOrder 0 -}}
+{{- $mockType := printf "Mock%sNodes" (title $leafClass) -}}
+
+package {{$instance.PackageInfo.GeneratedPackageAlias}}
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"time"
+
+
+	"github.com/uber/zanzibar/config"
+	"github.com/uber/zanzibar/runtime"
+
+	module "{{$instance.PackageInfo.ModulePackagePath}}"
+)
+
+// MockService interface
+type MockService interface {
+	MakeHTTPRequest(
+		method string,
+		url string,
+		headers map[string]string,
+		body io.Reader,
+	) (*http.Response, error)
+	MakeTChannelRequest(
+		ctx context.Context,
+		thriftService string,
+		method string,
+		headers map[string]string,
+		req, resp zanzibar.RWTStruct,
+	) (bool, map[string]string, error)
+	{{$mockType}}() *module.{{$mockType}}
+	Start()
+	Stop()
+}
+
+type mockService struct {
+	started        bool
+	server         *zanzibar.Gateway
+	{{camel $mockType}}    *module.{{$mockType}}
+	httpClient     *http.Client
+	tChannelClient zanzibar.TChannelCaller
+}
+
+// TestOptions is the info needed to create a test service
+type TestOptions struct {
+	TChannelClientMethods map[string]string
```

Comment:
Please add a line comment for this, since it's what the developer will inspect first to figure out how to pass information to the test runner.

Reply:
Totally agree that it is ambiguous and clueless. I actually want to completely hide it from tests because the way a user send request to the test service is via `MakeTChannelRequest` which takes the thrift service and method names from endpoint thrift idl, the tchannel client is not exposed and the `ZanzibarMethodName` is not used at all for whoever writes the test.

The `TChannelClientMethods` is used to create a client and the "ThriftService::method" to "ZanzibarMethodName" map is only useful for logs and metrics of that client, which can be occasionally helpful for debugging tests.

I will probably not exposed this at all or expose optionally. But the comment for the map is on point and I will put it where it should be.